baton/modules/twilio/twilio.go

```diff
@@ -0,0 +1,179 @@
+package twilio
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type Twilio struct {
+	UserId, ApiKey string
+}
+
+var (
+	smsCallbacks  = make([]callback, 0)
+	callCallbacks = make([]callback, 0)
+)
+
+type callback struct {
+	number string
+	resp   chan<- interface{}
+	data   string
+}
+
+var client = &http.Client{}
+
+func (t Twilio) RunCommand(cmd string, body interface{}, resp chan<- interface{}) error {
+	newBody := body.(map[string]interface{})
+	switch cmd {
+	case "send-sms":
+		return t.sendSMS(newBody, resp)
+	case "recieve-sms":
+		return t.recieveSMS(newBody, resp)
+	case "send-call":
+		return t.makeCall(newBody, resp)
+	case "recieve-call":
+		return t.recieveCall(newBody, resp)
+	default:
+		return errors.New("unknown command: " + cmd)
+	}
```

Comment:
Would it be cleaner to have a `map[string]func(map[string]interface{}, resp chan<- interface{})` and lookup the command in the map rather than a switch statement?